ui/helper.c

```diff
@@ -550,6 +550,59 @@ void ConvertRssiToDbm(uint16_t Rssi) {
 		}	
 }
 
+void ConvertRssiToSmeter(uint16_t Rssi)
+{
+	/* S-Meter to dbm
+	S9 + 40 dB 	-53 dBm
+	S9 + 30 dB 	-63 dBm
+	S9 + 20 dB 	-73 dBm
+	S9 + 10 dB 	-83 dBm
+	S9 			-93 dBm
+	S8 			-99 dBm
+	S7 			-105 dBm
+	S6 			-111 dBm
+	S5 			-117 dBm
+	S4 			-123 dBm
+	S3 			-129 dBm
+	S2 			-135 dBm
+	S1 			-141 dBm
+	S0 			-147 dBm
+	*/
+
+	int16_t RXdBM;
+	int16_t Svalue;
+	const int16_t RXdBMover9 = -93;
+	uint8_t len;
+
+	RXdBM = (Rssi >> 1) - 160;
+
+	for (int i = 0; i < 8; i++) {
+			gShortString[i] = ' ';
+		}
+
+	if (RXdBM <= RXdBMover9) {
+		Svalue = (-(-147 - RXdBM))/6; //6dBm spacing
+		if (Svalue < 0) Svalue = 0;
+		len = 1;
+	} else {
+		Svalue = -(RXdBMover9 - RXdBM);
+		if (Svalue > 99) Svalue = 99; //TODO: Delete or adapt this value on maximum value retrieved from BEKEN register
```

Comment:
Valid range for register is 72 ... 330 so max Svalue could be 217, but as there is a check for limit value in the code, it may be possible to get a value outside these range, so the real max of Svalue is 253...
IMO this line should remain...

Reply:
I agree with you.
Thanks